File: app/src/main/java/com/luhuan/rxsimple/TestActivity.java
package com.luhuan.rxsimple;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.util.Log;

import com.luhuan.rxsimple.dagger_computer.MacComputer;
import com.luhuan.rxsimple.dagger_computer.WinComputer;

import javax.inject.Inject;


public class TestActivity extends AppCompatActivity {

    @Inject
    UserName userName;

    @Inject
    AppleBean appleBean;

    @Inject
    @WinComputer
    ComputerBean winBean;

    @Inject
    @MacComputer
    ComputerBean macBean;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_test);
        DaggerTestComponent.builder().build().inject(this);
        Log.d("TAG", winBean.getType());
        Log.d("TAG", macBean.getType());
    }

}

File: app/src/main/java/com/luhuan/rxsimple/TestActivity.kt
package com.luhuan.rxsimple

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.util.Log

import com.luhuan.rxsimple.dagger_computer.MacComputer
import com.luhuan.rxsimple.dagger_computer.WinComputer

import javax.inject.Inject


class TestActivity : AppCompatActivity() {

    @Inject
    lateinit var userName: UserName

    @Inject
    lateinit var appleBean: AppleBean

    @Inject
    @field:WinComputer
    lateinit var winBean: ComputerBean

    @Inject
    @field:MacComputer
    lateinit var macBean: ComputerBean

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_test)
        DaggerTestComponent.create().inject(this)
        Log.d("TAG", winBean.type)
        Log.d("TAG", macBean.type)
    }

}
